src/languages/cpp.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  valueMatcher,
+} from "../nodeMatchers";
+import { NodeMatcherAlternative, ScopeType } from "../Types";
+
+// Generated by the following command:
+// >  curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-cpp/master/src/node-types.json | jq '[.[] | select(.type == "compound_statement") | .children.types[].type] + [.[] | select(.type == "_statement") | .subtypes[].type]'
+const STATEMENT_TYPES = [
+  "_statement",
+  "_type_specifier",
+  "alias_declaration",
+  "declaration",
+  "function_definition",
+  "linkage_specification",
+  "namespace_definition",
+  "preproc_call",
+  "preproc_def",
+  "preproc_function_def",
+  "preproc_if",
+  "preproc_ifdef",
+  "preproc_include",
+  "static_assert_declaration",
+  "template_declaration",
+  "template_instantiation",
+  "type_definition",
+  "using_declaration",
+  "break_statement",
+  "case_statement",
+  "compound_statement",
+  "continue_statement",
+  "do_statement",
+  "expression_statement",
+  "for_range_loop",
+  "for_statement",
+  "goto_statement",
+  "if_statement",
+  "labeled_statement",
+  "return_statement",
+  "switch_statement",
+  "throw_statement",
+  "try_statement",
+  "while_statement"
+];
+
+// >  curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-cpp/master/src/node-types.json | jq '[.[] | select(.type == "_type_specifier") | .subtypes[].type]'
+const TYPE_TYPES = [
+  "auto",
+  "class_specifier",
+  "decltype",
+  "dependent_type",
+  "enum_specifier",
+  "primitive_type",
+  "scoped_type_identifier",
+  "sized_type_specifier",
+  "struct_specifier",
+  "template_type",
+  "type_identifier",
+  "union_specifier"
+];
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  statement: STATEMENT_TYPES,
+  class: ["class_specifier", "struct_specifier", "enum_specifier", "union_specifier"],
+  className: [
+    "class_specifier[name]", "struct_specifier[name]", "enum_specifier[name]", "union_specifier[name]", 
+    "function_definition[declarator][declarator][namespace]" // void ClassName::method() {}
+  ],
+  ifStatement: "if_statement",
+  string: "string_literal",
+  comment: "comment",
+  arrowFunction: "lambda_expression",
+  list: "array_initializer",
+  functionCall: "call_expression",
+  name: ["*[declarator][declarator][name]", "*[declarator][name]", "*[declarator][declarator]",  "*[declarator]", "*[name]"],
+  namedFunction: "function_definition",
+  type: TYPE_TYPES.concat([ "*[type]" ]),
+  functionName: [
+    "function_definition[declarator][declarator][name]", // void C::funcName() {}
+    "function_definition[declarator][declarator]"        // void funcName() {}
+  ],
+  value: valueMatcher("*[declarator][value]", "*[value]"),
+  collectionItem: argumentMatcher("initializer_list"),
+  argumentOrParameter: argumentMatcher("parameter_list", "argument_list"),
+  xmlAttribute: "attribute"
```

Comment:
I guess it makes sense to rename it, since the voice command is just `attribute`.

Note that I don't have a real use case for selecting attributes -- I just went through the list of scopes in `cursorless help` and set up all that make sense for C++.

Reply:
If we want to use this same spoken word for other types of attributes I think we should rename the scope type.

Reply:
Filed in #198; fine to merge this one in without it